pkg/scheduler/scheduler.go

```diff
@@ -112,31 +114,55 @@ func (rs *RemoteScheduler) schedule(state cluster.State) {
 		return
 	}
 
+	//Set the status as reconciling
+	_, err = rs.inventory.UpdateStatus(&state, model.Reconciling)
+	if err != nil {
+		rs.logger.Infof("Failed to update cluster status as reconciling: %s", err)
+	}
+
 	//Reconcile CRD components first
 	for _, component := range components {
 		if rs.isCRDComponent(component.Component) {
-			rs.reconcile(component, state, schedulingID, doInstallCRD, concurrencyNotAllowed)
+			rs.reconcile(nil, component, state, schedulingID, doInstallCRD, concurrencyNotAllowed)
 		}
 	}
 
 	//Reconcile pre components
 	for _, component := range components {
 		if rs.isPreComponent(component.Component) {
-			rs.reconcile(component, state, schedulingID, doNotInstallCRD, concurrencyNotAllowed)
+			rs.reconcile(nil, component, state, schedulingID, doNotInstallCRD, concurrencyNotAllowed)
 		}
 	}
 
+	var wg sync.WaitGroup
+
 	//Reconcile the rest
 	for _, component := range components {
 		if rs.isPreComponent(component.Component) || rs.isCRDComponent(component.Component) {
 			continue
 		}
-		rs.reconcile(component, state, schedulingID, doNotInstallCRD, concurrencyAllowed)
+		rs.reconcile(&wg, component, state, schedulingID, doNotInstallCRD, concurrencyAllowed)
+	}
+
+	wg.Wait()
+	//Check if any component reconciliation errored, if not, set the status as ready
+	latestState, err := rs.inventory.GetLatest(state.Cluster.Cluster)
+	if err != nil {
+		rs.logger.Infof("Failed to update cluster status as reconciling: %s", err)
+	}
+	if latestState.Status.Status != model.Error {
+		_, err = rs.inventory.UpdateStatus(&state, model.Ready)
+		if err != nil {
+			rs.logger.Infof("Failed to update cluster status as ready: %s", err)
+		}
 	}
 }
 
-func (rs *RemoteScheduler) reconcile(component *keb.Components, state cluster.State, schedulingID string, installCRD bool, concurrent concurrency) {
-	fn := func(component *keb.Components, state cluster.State, schedulingID string) {
+func (rs *RemoteScheduler) reconcile(wg *sync.WaitGroup, component *keb.Components, state cluster.State, schedulingID string, installCRD bool, concurrent concurrency) {
+	fn := func(wg *sync.WaitGroup, component *keb.Components, state cluster.State, schedulingID string, concurrent concurrency) {
+		if bool(concurrent) {
```

Comment:
as above.. maybe we use the `wg` as indicator for concurrency..

e.g. 

     func useConcurreny(wg *sync.WaitingGroup) bool {
         return wg != nil
     }